src/store/persona/slice.ts

```diff
@@ -0,0 +1,61 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { initialState } from 'store/persona/types';
+import { createPersonaUser, fetchPersonaUser } from 'store/persona/thunks';
+
+// Since api can return null for personaUserInfo, undefined must be set by default
+export const PersonaState: initialState = {
+  personaUserInfo: undefined,
+  isLoading: true,
+  isUpdating: false,
+  isDeleting: false,
+};
+
+const personaSlice = createSlice({
+  name: 'persona',
+  initialState: PersonaState,
+  reducers: {
+    cleanLogout: (state) => {
+      return PersonaState;
+    },
+    setIsPersonaLoading: (state, action: PayloadAction<boolean>) => ({
+      ...state,
+      isLoading: action.payload,
+    }),
+  },
+  extraReducers: (builder) => {
+    // Create Persona User
```

Comment:
comment not really helpful in my opinion (same thing for similar ones below)

Reply:
The comment is used to delimite all the call (pending, fullfilled, rejected). Whitout them, it take more time to see which call are for creating, fetching, updating

```
  extraReducers: (builder) => {
    builder.addCase(createPersonaUser.pending, (state) => ({
      ...state,
      isLoading: true,
      error: undefined,
    }));
    builder.addCase(createPersonaUser.fulfilled, (state, action) => ({
      ...state,
      isLoading: false,
      personaUserInfo: action.payload,
    }));
    builder.addCase(createPersonaUser.rejected, (state, action) => ({
      ...state,
      error: action.payload,
      isLoading: false,
    }));
    builder.addCase(fetchPersonaUser.pending, (state) => {
      state.isLoading = true;
      state.error = undefined;
    });
    builder.addCase(fetchPersonaUser.fulfilled, (state, action) => ({
      ...state,
      isLoading: false,
      personaUserInfo: action.payload,
    }));
    builder.addCase(fetchPersonaUser.rejected, (state, action) => ({
      ...state,
      error: action.payload,
      isLoading: false,
    }));
  },
```